promql/ast.go

```diff
@@ -79,11 +82,12 @@ type Expressions []Expr
 
 // AggregateExpr represents an aggregation operation on a Vector.
 type AggregateExpr struct {
-	Op       ItemType // The used aggregation operation.
-	Expr     Expr     // The Vector expression over which is aggregated.
-	Param    Expr     // Parameter used by some aggregators.
-	Grouping []string // The labels by which to group the Vector.
-	Without  bool     // Whether to drop the given labels rather than keep them.
+	Op            ItemType // The used aggregation operation.
+	Expr          Expr     // The Vector expression over which is aggregated.
+	Param         Expr     // Parameter used by some aggregators.
+	Grouping      []string // The labels by which to group the Vector.
+	Without       bool     // Whether to drop the given labels rather than keep them.
+	positionRange PositionRange
```

Comment:
These should probably be public for the sake of 3rd party users of this code.

Reply:
The position data can be read accessed trough the public `PositionRange` interface function.

Are there any 3rd parties that want write access this?

Reply:
I don't know, but probably doesn't hurt to let them considering they can write everything else.

Reply:
Well, only almost everything.

https://github.com/prometheus/prometheus/blob/201491fd1b6b02fab60f7630dff23218bc3d1067/promql/ast.go#L152

Reply:
That's part of the engine implementation, not the AST.

Reply:
I've made them public.